our_site/planner/views.py

```diff
@@ -263,3 +265,30 @@ def invitations(request, event_id: int) -> HttpResponse:
             request,
         )
     )
+
+class RoleDetails(forms.Form):
+    role_name=forms.CharField(max_length=30)
+    #description= forms.CharField(max_length=100)
+    #event=forms.ForeignKey(Event, on_delete=models.CASCADE, related_name='roles')
+    #amount = forms.IntegerField()
+
+def addRoles(request):
+    if request.user.is_anonymous:
+        return HttpResponseRedirect("/account/login/")
+    if request.method=="GET":
+        template=loader.get_template("planner/addRoles.html")
+        addRoles_form=RoleForm()
+        return HttpResponse(template.render({"addRoles_form": addRoles_form},request))
+
+    elif request.method=="POST":
+        addRoles_form=RoleForm(request.POST)
+        if addRoles_form.is_valid():
+            role=addRoles_form.save(commit=False)
+            role.user=request.user
+            role.save()
+
+        addRoles_form=RoleForm()
+        return render(request, "planner/dashboard.html", {"addRoles_form": addRoles_form})
+    else:
```

Comment:
What exactly is happening in this case? It seems to present a severely shortened version of the role form? But the user can't do anything with that form because there's no backend interaction for it, no? Is the class RoleDetails really necessary?